### PR TITLE
feat: Implement `plotly.HTMLDataset`

### LINKED ISSUE #787: `kedro-datasets`: Implement `plotly.HTMLDataset`
## Description

Currently both of existing `plotly` datasets save a figure **as JSON** and load from it.

This has a major limitation because **viewing them requires running Python** to parse it back as `plotly` object. The only options are `kedro viz` or `catalog.load("figure_dataset")`.

I think there is value in having a `plotly.HTMLDataset` that would save figures on disk as HTML so:

- They can be opened without the need to run Python.
- It would also make them shareable with team members who don't have Kedro project setup.
- Using `versioned: True` in catalog you'd get a convenient folder of HTML files that you'd be able to scroll through without running Python / Kedro.

## Context
Sharing charts with teammates is a task I often encounter and it would be nice if I could share `plotly` charts as HTMLs rather than having to run `kedro viz` or `catalog.load("figure_dataset")` and doing a screenshot manually.

## Possible Implementation

- See [this minimal project](https://github.com/yury-fedotov/kedro-plotly-html) where I did a proof-of-concept implementation.
- Dataset implementation is [here](https://github.com/yury-fedotov/kedro-plotly-html/blob/main/src/kedro_plotly_html/kedro_datasets/plotly/html_dataset.py).

### PR DESCRIPTION
**Pull request by yury-fedotov**
## Description
Closes #787 

## Development notes

- Given that this dataset is **very** similar to existing `plotly.JSONDataset`, I've added it as core. Though let me know if you prefer it to be experimental.
- I tried to keep the test file as similar to `plotly.JSONDataset` as possible.

There are just two differences with `plotly.JSONDataset`:

- This one cannot `load` at all because `plotly` doesn't support anything like `read_html`.
- This one is using `plotly.io.write_html` while the other one is using `plotly.io.write_json`.

## Checklist

- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the relevant `RELEASE.md` file
- [x] Added tests to cover my changes

### CODE REVIEW

**Review by merelcht — APPROVED** [review]
Thanks for the contribution @yury-fedotov ! I'm happy to get this in as a regular dataset. It is indeed very similar to `plotly.JSONDataset` so I don't see a significant maintenance burden for the Kedro team to support this dataset.

We'll need 1/2 approval from the TSC to merge this dataset to the core datasets space. As of today we have 20 TSC members so we'll need 10 approvals in total.

**Review by astrojuanlu — APPROVED** [review]
Thanks a lot for your contribution @yury-fedotov ! 🙌🏼

**Review by ravi-kumar-pilla — APPROVED** [review]
Thank you @yury-fedotov

**Review by tynandebold — APPROVED** [review]
Excellent! Thank you!

File: kedro-datasets/pyproject.toml [hunk 1]
```diff
@@ -121,8 +121,9 @@ pillow-imagedataset = ["Pillow~=9.0"]
 pillow = ["kedro-datasets[pillow-imagedataset]"]
 
 plotly-jsondataset = ["kedro-datasets[plotly-base]"]
+plotly-htmldataset = ["kedro-datasets[plotly-base]"]
 plotly-plotlydataset = ["kedro-datasets[pandas-base,plotly-base]"]
-plotly = ["kedro-datasets[plotly-jsondataset,plotly-plotlydataset]"]
+plotly = ["kedro-datasets[plotly-jsondataset,plotly-htmldataset,plotly-plotlydataset]"]
```
Comment (deepyaman) [inline kedro-datasets/pyproject.toml:126]: Same comment on alphabetization.

File: kedro-datasets/kedro_datasets/plotly/html_dataset.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+
+    def _describe(self) -> dict[str, Any]:
+        return {
+            "filepath": self._filepath,
+            "protocol": self._protocol,
+            "save_args": self._save_args,
+            "version": self._version,
+        }
+
+    def _load(self) -> NoReturn:
```
Comment (deepyaman) [inline kedro-datasets/kedro_datasets/plotly/html_dataset.py:130]: Nice, didn't actually know of `NoReturn`

**Review by deepyaman — APPROVED** [review]
Going to approve, but have a few nit comments around alphabetization I'd appreciate. :D

**Review by Galileo-Galilei — APPROVED** [review]

**Review by noklam — APPROVED** [review]

**Review by rashidakanchwala — APPROVED** [review]
thanks @yury-fedotov

**Review by ElenaKhaustova — APPROVED** [review]
Thank you, @yury-fedotov!

**Review by jitu5 — APPROVED** [review]
Thanks!

**Comment by merelcht** [comment]
This has 12/20 TSC approval so I'm going to merge it. Thanks again for the contribution @yury-fedotov ⭐

**Pull request merged**